src/validation.cpp

```diff
@@ -4645,7 +4641,7 @@ static const uint64_t MEMPOOL_DUMP_VERSION = 1;
 
 bool LoadMempool(CTxMemPool& pool, CChainState& active_chainstate, FopenFn mockable_fopen_function)
 {
-    int64_t nExpiryTimeout = gArgs.GetIntArg("-mempoolexpiry", DEFAULT_MEMPOOL_EXPIRY) * 60 * 60;
+    int64_t nExpiryTimeout = std::chrono::seconds{pool.m_expiry}.count();
```

Comment:
nit in e741c49716a65b624fcf91db0de6dd11617675ad: Might be nice to make `nExpiryTimeout` type-safe in a follow-up.

Reply:
True! Not addressing in this PR.